app/jobs/webhook/resource_fetch_job.rb

```diff
@@ -3,14 +3,21 @@ class Webhook::ResourceFetchJob < ApplicationJob
 
   def perform(webhook_receipt)
     peer = webhook_receipt.peer
-    res = Tor::HTTP.get(peer.onion_address, "/api/v1/webhook/#{webhook_receipt.token}.json")
+    res = Tor::HTTP.get(peer.onion_address, "/api/v1/webhook/#{webhook_receipt.uuid}/#{webhook_receipt.token}.json")
     response = JSON.parse(res.body)
-    resource = resource_type(response).create!(
-      response['resource']
+    if response['action'] == 'upsert'
+      resource = resource_type(response)
+                 .find_or_initialize_by(peer: peer, uuid: response['uuid'])
```

Comment:
needs to see if another peer has already claimed UUID, another peer may be trying to tell about this one.

leave todo note here about maybe allowing updates if original peer has been offline long enough, essentially re-routing propagations